src/basic/fd-util.c

```diff
@@ -930,3 +930,27 @@ int fd_reopen(int fd, int flags) {
 
         return new_fd;
 }
+
+int read_nr_open(void) {
+        _cleanup_free_ char *nr_open = NULL;
+        int r;
+
+        /* Returns the kernel's current fd limit, either by reading it of /proc/sys if that works, or using the default
+         * compiled-in version of the kernel if not */
```

Comment:
That second part of the sentence doesn't make sense: "compiled-in version of the kernel".